lib/ipni-client.js

```diff
@@ -1,12 +1,15 @@
 import { decodeBase64, decodeVarint, pRetry, assertOkResponse } from '../vendor/deno-deps.js'
 
+/** @typedef {{ address: string; protocol: string; contextId: string; }} Provider  */
+
 /**
  *
  * @param {string} cid
  * @param {string} providerId
  * @returns {Promise<{
  *  indexerResult: string;
- *  provider?: { address: string; protocol: string };
+ *  provider?: Provider;
+ *  providers?: Provider[];
```

Comment:
I find it confusing to have two properties called `provider` and `providers`, I cannot tell what's the difference.

How about calling the new property `alternativeProviders` or `altProviders`?